### ISSUE TITLE
MailChimp plugin emits NOTICE

### ISSUE THREAD

**Issue by calevans** [mautic/mautic#4381]
What type of report is this:

| Q  | A
| ---| ---
| Bug report? | X
| Feature request? | 
| Enhancement? | 

## Description:
The Mailchimp plugin emits the following notice anytime it is run.

PHP Notice:  Undefined index: error in /public_html/plugins/MauticEmailMarketingBundle/Api/MailchimpApi.php on line 54)

## If a bug:

| Q   | A
| --- | ---
| Mautic version | 2.9 BETA 
| PHP version | 7.1

### Steps to reproduce:
1. php /public_html/app/console mautic:messages:send
2. 
 
### Log errors: 
Nothing in the Mautic log files. The above error is otput to my cron log.

**Comment by calevans** [comment]
UPDATE

This is more serious than I thought. The notice is just a symptom of the problem. The mailchimp response does contain an errors index, but within it, it odes not. The one that I am getting back looks like this when var_dumped:
    [errors] => Array
        (
            [0] => Array
                (
                    [field] => merge_fields
                    [message] => Schema describes object, array found instead
                )

        )

)

So you see that errors does not contain an index called errors but on line 54 of MailchimpApi.php  it is trying to use it. Since it does not exist, it just throws the notice and keeps moving, but the bigger issue is that we are not actually getting the errors. back. 

Cheers!
=C=

**Comment by mqueme** [comment]
would this be a Mailchimp error? or a Mandrill error? Mandrill is the mail service.

**Comment by calevans** [comment]
This is an error coming from Mailchimp. I do not use Mandril.

Cheers! :)
=C=

**Comment by calevans** [comment]
The full payload returned from Mailchimp is:

```
Array
(
    [type] => http://developer.mailchimp.com/documentation/mailchimp/guides/error-glossary/
    [title] => Invalid Resource
    [status] => 400
    [detail] => The resource submitted could not be validated. For field-specific details, see the 'errors' array.
    [instance] =>
    [errors] => Array
        (
            [0] => Array
                (
                    [field] => merge_fields
                    [message] => Schema describes object, array found instead
                )

        )

)
```
If that helps.

I put in #4383 but it applies here too.

Cheers! :)
=C=

**Comment by calevans** [comment]
Fixed with #4455 

=C=
